frontend/src/app/core/services/http.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpResponse } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs';
+@Injectable({
+  providedIn: 'root',
+})
+export class HttpService {
+  public baseUrl: string = environment.apiUrl;
+  public headers = new HttpHeaders();
+
+  constructor(private readonly http: HttpClient) {}
+  public getHeaders(): HttpHeaders {
```

Comment:
Could you omit access modificator since it's public by default and don't write return type if it can be taken from context automatically? Apply those rules for TS code only